src/vmm/src/resources.rs

```diff
@@ -225,6 +225,7 @@ impl VmResources {
             vcpu_count: self.vm_config().vcpu_count,
             smt: self.vm_config().smt,
             cpu_template: self.vm_config().cpu_template,
+            custom_cpu_template: self.vm_config().custom_cpu_template.clone(),
```

Comment:
Why separate `cpu_tempate` and `custom_cpu_template`?

Reply:
CpuFeaturesTemplate requires versionization as it is a part of the snapshot VmInfo type. However the custom CPU template should not be serialized by the snapshot.

Reply:
+1. The following implementation looks simpler to me.
```rs
pub enum CpuFeaturesTemplate {
    /// Custom template
    CUSTOM(CustomCpuTemplate),
...
}
```